libnetwork/ipam/structures.go

```diff
@@ -46,6 +55,14 @@ type addrSpace struct {
 // PoolIDFromString creates a new PoolID and populates the SubnetKey object
 // reading it from the given string.
 func PoolIDFromString(str string) (pID PoolID, err error) {
+	if strings.HasPrefix(str, poolIDV2Prefix) {
```

Comment:
Yeah, so that's one reason. The other reason is that `poolIDV2Prefix` identifies the format version and then it's up to the decoder to use its own codec. This makes `PoolIDFromString` unaware of implementation details -- every decoder is self-contained. I think that's cleaner if we ever want to make a v3 format.

Also, this is a backport so I think it's best to keep it as is.